### PR TITLE
modernised python packaging

### PR DESCRIPTION
**Pull request by batmanfeynman**
I have made a pyproject.toml file as well as a setup.cfg file which is compliant with the latest python packaging recommendations.
These changes help me in creating an rpm for fedora as well. I will make an rpm package for v1.6.9 by including a patch file, but i think this modernized python packaging will help in possible future installation from pip.

### CODE REVIEW

**Comment by daniel-123** [comment]
AppImage seems to work just fine on my Debian and this doesn't touch any of the deb packages, so from my side that seems to be covered.

One thing that isn't is that your new method assumes that end user will always want the GUI. Which is reasonably fair assumption on a client, but not so much in case of headless servers. Do you have some ideas how to have an optional set of requirements? Preferably, but not necessarily while keeping the GUI libraries as default.

**Comment by batmanfeynman** [comment]
The changes introduced in #416 is what i came up with basically (those changes can easily be translated to the setup.cfg file). But that results in the following setup:
`pip install syncplay` would install everything without the gui requirements. 
`pip install syncplay[GUI]` would install the client gui as well.

i noticed that @FallenWarrior2k head searched for a way to make the GUI default and couldn't find one. I am inclined to believe them. But I will look around a bit more to see if it possible without any workarounds.

**Comment by FallenWarrior2k** [comment]
I believe the extras-based approach is blocked on pypa/setuptools#1139.

**Comment by batmanfeynman** [comment]
Also to add some information, I tried to see if i could use a venv with pip to run the syncplay code on both a fresh ubuntu and Fedora VM.
Ubuntu:
Installing requirements with pip went well and it fetched pyside2 5.15.2. All good. But syncplay client wouldn't run and complained about some missing library. I had to install some package through apt to get it working from within the venv

Fedora:
I made a venv. pip install for some reason only installed pyside2 5.13.1 . This was throwing some error on the python code of syncplay. Installing the python-packages from fedora repos did not get rid of this error! But now that all the dependencies were installed, if i exit the venv, syncplay would work.

The reason i am pointing out my experience is to show that getting gui working with pip based install method probably wont be straighforward.

**Comment by batmanfeynman** [comment]
But I do agree with @FallenWarrior2k in that having at least the headless server package easily available on pypi would be awfully convenient.

Maybe we can make just the server package available through PyPI?

**Comment by batmanfeynman** [comment]
@daniel-123 I also want to point out that with this commit, all i have done is translate the old setup.py file to the new format. The behaviour is the same, as in the old setup.py file also assumed that the end user will always want the GUI client.

What I want to point out is that making syncplay avaialable on PyPI is not the primary objective of this pull request, but instead to make it easier to follow packaging guidelines in fedora for projects developed in python. I will make a copr repo by tomorrow for 1.6.9 and link it here

**Comment by albertosottile** [comment]
Before staring the review, just FYI, PySide2 wheels contain their own copy of Qt. So, in principle, there should be no need to resort to `apt` to run Syncplay after installing it from Python sources. However, there are some combinations of OS-PySide2 versions for which **other** libraries are missing when doing so. 

This is the case for Ubuntu 20.04.3 and PySide2 5.15.2, in which the library missing is `libxcb-xinerama.so.0` (see https://askubuntu.com/a/1069502). I just reproduced this on a fresh Focal Fossa VM. IMHO, this is a bug that should be reported upstream to PySide2. However, I can see that they cannot possibly support every target available... (though they should keep an eye on Ubuntu). 

In short, I would not consider impossible to install Syncplay with GUI from PyPi, provided that all the dependencies behave as they should. Nevertheless, I do not think that PyPI should be a supported distribution channel for this project, as we already have quite a few. We actually invest a lot of effort for providing binaries specifically prepared to avoid having to use the sources.

In addition, I do not see the benefits of PyPi over plain `git` for developers and enthusiasts that still want to run from sources, as Syncplay cannot be used as a dependency in other Python projects.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    Operating System :: Microsoft :: Windows
+    Operating System :: POSIX :: Linux
+    Natural Language :: English
+    Natural Language :: German
+    Natural Language :: Italian
+    Natural Language :: Russian
+    Natural Language :: Spanish
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.4
+    Programming Language :: Python :: 3.5
```
Comment (albertosottile) [inline setup.cfg:33]: Same for the 3.4 and 3.5 classifiers, they should be deleted.

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+include_package_data = True
+packages = find:
+install_requires =
+    twisted[tls]>=16.4.0
+    certifi>=2018.11.29
+    appnope>=0.1.0; sys_platform == 'darwin'
+    pypiwin32>=223; sys_platform == 'win32'
+    zope.interface>=4.4.0; sys_platform == 'win32'
+    pyside2>=5.11.0
+    requests>=2.20.0; sys_platform == 'darwin'
```
Comment (albertosottile) [inline setup.cfg:53]: Could you put these in alphabetical order?

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Programming Language :: Python :: 3 :: Only
+    Topic :: Internet
+    Topic :: Multimedia :: Video
+
+[options]
+include_package_data = True
+packages = find:
+install_requires =
```
Comment (albertosottile) [inline setup.cfg:46]: I am not a fan of duplicated information, and switching to PEP 621 without touching the `requirements*.txt` files does exactly that. 

I would be in favor of deleting the requirement files, but currently the Windows build system relies on them. This could be circumvented by using `pip-compile` e.g. via
```
pip-compile --output-file requirements.txt setup.cfg
```
in the CI workflows before the file is actually needed.

On the other hand, we could retain the requirements files and start using them in the proper way: to pin dependencies for CI/release builds. We actually need to pin dependencies on macOS but, for the moment I did that in the workflow code (very bad, but I did not have a proper alternative up to now).

@Et0h @daniel-123 what is your opinion here?

**Review by albertosottile — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 3 above)
Comment (Et0h) [reply setup.cfg:46]: > @Et0h @daniel-123 what is your opinion here?

As someone who is a Windows user I don't have any strong opinions on packaging beyond the general desire for things to work and be low-maintenance.

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    twisted[tls]>=16.4.0
+    certifi>=2018.11.29
+    appnope>=0.1.0; sys_platform == 'darwin'
+    pypiwin32>=223; sys_platform == 'win32'
+    zope.interface>=4.4.0; sys_platform == 'win32'
+    pyside2>=5.11.0
+    requests>=2.20.0; sys_platform == 'darwin'
+python_requires = >=3.4
```
[the remark this answers is not in the document]
Comment (batmanfeynman) [reply setup.cfg]: done

File: setup.cfg (hunk 2 above)
Comment (batmanfeynman) [reply setup.cfg]: done

File: setup.cfg (hunk 1 above)
Comment (batmanfeynman) [reply setup.cfg]: done

**Comment by batmanfeynman** [comment]
Link to copr repo as promised:
https://copr.fedorainfracloud.org/coprs/batmanfeynman/syncplay/

I also want to suggest 3 modification to the syncplay.pl website:
1.  In [this page](https://syncplay.pl/guide/trouble/) under "Problems running Appimage of Syncplay on Fedora", the link to issue number #355 is not present. Please fix this.
2. Add the link to the copr page at the end of [this page](https://syncplay.pl/download/)
3. Maybe suggest using the rpm in my copr instead of the AppImage for now in the troubleshooting page

File: setup.cfg (hunk 3 above)
Comment (daniel-123) [reply setup.cfg:44]: Personally I'd prefer to avoid pinning dependencies to specific version unless strictly as rollback resulting from a bug in latest version of given dependency. At least as far as AppImage goes.

For OS packages - they just use what's on the OS, so pinning isn't really a useful concept in first place.

Comment (batmanfeynman) [reply setup.cfg:44]: Hi @daniel-123 @albertosottile 
I just wanted to remind you of this!

**Comment by soredake** [comment]
@batmanfeynman can you please add support for fedora 36 in your repo?

**Comment by batmanfeynman** [comment]
@soredake done!

**Comment by corbmr** [comment]
@batmanfeynman Would it possible to get fedora 37 added to your repo? Thank you for working on this!

**Comment by batmanfeynman** [comment]
@corbmr  done.

File: setup.cfg (hunk 3 above)
Comment (albertosottile) [reply setup.cfg:44]: Sorry if I neglected this. The gist of this issue is that in both Windows and macOS workflows we currently have:
```
          pip3 install -r requirements.txt
          pip3 install -r requirements_gui.txt
```

and I do not see a clear way to reproduce the same behavior after this PR is merged UNLESS the information about dependencies is duplicated in `setup.cfg` and in the two requirements file. Am I missing something here? Is there a command to just install the dependencies of a package?

**Comment by albertosottile** [comment]
General comment: the issue of having the GUI dependencies separate is not solved. I believe the vast majority of people that install from sources do that precisely because they do not want to have the GUI. Therefore, I do not think we can just merge this PR in a way that does not cover for this use case. Unless I am missing something, I think we need to keep the two separate requirements file, for the moment.

**Comment by Et0h** [comment]
> Link to copr repo as promised: https://copr.fedorainfracloud.org/coprs/batmanfeynman/syncplay/
> 
> I also want to suggest 3 modification to the syncplay.pl website:
> 
> 1. In [this page](https://syncplay.pl/guide/trouble/) under "Problems running Appimage of Syncplay on Fedora", the link to issue number [PySide missing from AppImage of version 1.6.5 #355](https://github.com/Syncplay/syncplay/issues/355) is not present. Please fix this.

Fixed.

> 2. Add the link to the copr page at the end of [this page](https://syncplay.pl/download/)

I've added it to https://copr.fedorainfracloud.org/coprs/batmanfeynman/syncplay/ and https://github.com/Syncplay/syncplay/discussions/574

> 3. Maybe suggest using the rpm in my copr instead of the AppImage for now in the troubleshooting page

Done.

**Comment by VarLad** [comment]
@Et0h @batmanfeynman Any idea if you could update the COPR Fedora 38 (and possibly build for Rawhide) and update the version to the latest release?
